tap_kafka/sync.py

```diff
@@ -164,8 +196,24 @@ def bookmarked_partition_to_next_position(topic: str,
         raise InvalidBookmarkException(f"Invalid bookmark. One or more bookmark entries using invalid type(s).")
 
 
-def assign_consumer_to_bookmarked_state(consumer, topic: str, state, assign_by: str = 'timestamp'):
-    """Asssign consumer to bookmarked positions"""
+def assign_consumer_to_timestamp(consumer, topic: str, timestamp: str, timeout: int = 30) -> None:
+    """Assign consumer topic of all partitions to given timestamp"""
+    # Get list of all partitions
+    cluster_md = consumer.list_topics(topic=topic, timeout=timeout)
+    topic_md = cluster_md.topics[topic]
+    partitions = topic_md.partitions
+
+    # Find the offset of partitions by timestamps
+    epoch = iso_timestamp_to_epoch(timestamp)
+    partitions_list = [confluent_kafka.TopicPartition(topic, p, epoch) for p in partitions]
+    partitions_to_assign = consumer.offsets_for_times(partitions_list)
+
+    # Assign to the correct position
+    consumer.assign(partitions_to_assign)
+
+
+def assign_consumer_to_bookmarked_state(consumer, topic: str, state, assign_by: str = 'timestamp') -> None:
+    """Assign consumer to bookmarked positions"""
     bookmarked_partitions = state.get('bookmarks', {}).get(topic, {})
```

Comment:
this function is for `assign_consumer_to_bookmarked_state` then we must expect the state dictionary to have bookmark per topic, so we should instead use brackets to access them: `state['bookmarks'].[topic]`

Reply:
thanks fixed